src/test/kotlin/blackjack/domain/player/PlayerStateTest.kt

```diff
@@ -0,0 +1,104 @@
+package blackjack.domain.player
+
+import blackjack.domain.card.Card
+import blackjack.domain.card.CardSuit
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class PlayerStateTest {
+    @Test
+    fun `숫자 합이 21 미만이라면 게임 진행 상태가 된다`() {
+        val player = Player("vivian")
+        val cardsWithTotalBelowTwentyOne = listOf(
+            Card.Ten(CardSuit.CLOVER),
+            Card.Ten(CardSuit.HEART)
+        )
+
+        cardsWithTotalBelowTwentyOne.forEach { player.addCardToHand(it) }
+
+        assertThat(PlayerState.of(player)).isExactlyInstanceOf(PlayerState.Playing::class.java)
+    }
+
+    @Test
+    fun `숫자 합이 21 이라면 게임 진행 불가 상태가 된다`() {
+        val player = Player("vivian")
+        val cardsWithTotalBelowTwentyOne = listOf(
+            Card.Ace(CardSuit.CLOVER),
+            Card.Ten(CardSuit.HEART)
+        )
+
+        cardsWithTotalBelowTwentyOne.forEach { player.addCardToHand(it) }
+
+        assertThat(PlayerState.of(player)).isExactlyInstanceOf(PlayerState.Done::class.java)
+    }
+
+    @Test
+    fun `숫자 합이 21 초과라면 게임 진행 불가 상태가 된다`() {
+        val player = Player("vivian")
+        val cardsWithTotalBelowTwentyOne = listOf(
+            Card.Ten(CardSuit.CLOVER),
+            Card.Ten(CardSuit.HEART),
+            Card.Ten(CardSuit.DIAMOND)
+        )
+
+        cardsWithTotalBelowTwentyOne.forEach { player.addCardToHand(it) }
+
+        assertThat(PlayerState.of(player)).isExactlyInstanceOf(PlayerState.Done::class.java)
+    }
+
+    @Test
+    fun `사용자가 게임을 진행할 수 있는 상태이고 hit 를 해 카드를 받은 후 총 숫자합이 21 미만일 경우 계속 게임 진행 상태로 유지된다`() {
+        val player = Player("vivian")
+
+        val cards = listOf(
+            Card.Four(CardSuit.CLOVER),
+            Card.Six(CardSuit.HEART),
+        )
+        val additionalCard = Card.King(CardSuit.CLOVER)
+
+        cards.forEach { player.addCardToHand(it) }
+
+        val playerState = PlayerState.of(player)
+        assertThat(playerState).isExactlyInstanceOf(PlayerState.Playing::class.java)
+
+        val newPlayerState = (playerState as PlayerState.Playing).hit(additionalCard)
+        assertThat(newPlayerState).isExactlyInstanceOf(PlayerState.Playing::class.java)
+    }
+
+    @Test
+    fun `사용자가 게임을 진행할 수 있는 상태이고 hit 를 해 카드를 받은 후 총 숫자합이 21 이상일 경우 게임 불가 상태가 된다`() {
+        val player = Player("vivian")
+
+        val cards = listOf(
+            Card.Six(CardSuit.CLOVER),
+            Card.Six(CardSuit.HEART),
+        )
+        val additionalCard = Card.King(CardSuit.CLOVER)
+
+        cards.forEach { player.addCardToHand(it) }
+
+        val playerState = PlayerState.of(player)
+        assertThat(playerState).isExactlyInstanceOf(PlayerState.Playing::class.java)
+
+        val newPlayerState = (playerState as PlayerState.Playing).hit(additionalCard)
+        assertThat(newPlayerState).isExactlyInstanceOf(PlayerState.Done::class.java)
+    }
```

Comment:
아래 코드로 개선해 볼 수 있을까요?
```suggestion
    @Test
    fun `22점 이상이 되는 hit 를 하면 Done 을 리턴한다`() {
        val playing = Playing(
            Six(CLOVER), Six(HEART)
        )

        assertThat(playing.hit(King(CLOVER))).isEqualTo(Done)
    }
```
- 불필요한 중복을 줄였어요. (`PlayerState.of(player)` 의 결과가 `Playing` 클래스라는건 다른 테스트에 이미 존재해요)
- 가짜 생성자를 이용해 테스트를 위한 상태를 만드는 과정을 최대한 생략했어요. (보여주고 싶은건 
- 생성자 주입을 이용해 테스트를 위해 필요한 상태를 바로 만들었어요. (한번에 하나만 테스트 해요. `addCardToHand` 가 잘 동작하는지는 다른 테스트에서 검증해요)
- 테스트 이름에 무엇을 테스트 하려는지 직관적으로 표현해요. 
> 좋은 테스트 이름
> - 어떤 동작을 하면 어떤 결과가 나온다
> - 어떤 결과는 어떤 조건에서 발생한다
> - 어떤 일을 하면 어떤 결과가 나온다
- 코드또한 사람이 읽음을 고려해서 static import를 활용, 최대한 간결하게 표현해요. 

다른 테스트들도 이와같이 개선할 부분이 있을지 고민해 보세요~